### PR TITLE
Added ability to give sidekicks the ability to use an embedded application

### PR DESCRIPTION
**Pull request by bradtaylorsf**
This pull request introduces a new feature that allows users to choose between a traditional chatbot interface and an embedded form for chatflows. The changes include updates to the frontend UI, backend services, and database schema to support this new functionality.

Key Changes:
1. Frontend (GeneralSettings.jsx):
Added a new "Display Mode" dropdown to select between "Chatbot" and "Embedded Form".
Implemented an "Embedded URL" input field that appears when "Embedded Form" is selected.
Updated the save functionality to include the new display mode and embedded URL in the chatbotConfig.

2. Backend (chatflows/index.ts):
Modified the updateChatflow function to handle the new chatbotConfig field, which includes displayMode and embeddedUrl.
Updated error handling and logging for better debugging.
Database Schema (Interface.ts):
Added displayMode and embeddedUrl to the IChatFlow interface.

4. Chat Detail Component (ChatDetail.tsx):
Implemented conditional rendering based on the selected display mode.
Added an iframe to display the embedded form when the "Embedded Form" mode is selected.
5. Documentation (general.md):
Updated the documentation to reflect the new display mode and embedded form features.
6. Package Information:
Updated package name and version in package.json.

Testing:
Tested the new display mode selection in the GeneralSettings component.
Verified that the embedded form displays correctly when selected.
Ensured that the chatbot interface still functions as expected when in chatbot mode.
Checked that the new configuration is correctly saved to and retrieved from the database.

Notes:
This feature allows for greater flexibility in how chatflows can be presented to end-users.
Care should be taken when using the embedded form feature to ensure the embedded URL is from a trusted source.

### CODE REVIEW

File: packages-answers/ui/src/ChatDetail.tsx [hunk 1]
```diff
@@ -171,61 +175,73 @@ export const ChatDetail = ({
...
+                                                <MessageCard
+                                                    {...message}
+                                                    key={`message_${index}`}
+                                                    setSelectedDocuments={setSelectedDocuments}
+                                                />
+                                            ))}
 
-                                    {error ? (
-                                        <>
-                                            <MessageCard id='error' role='assistant' content={`${error.message} `} error={error} />
```
Comment (maxtechera) [inline packages-answers/ui/src/ChatDetail.tsx:189]: Not sure if we have support for this role as it's not openai valid

File: packages/docs/docs/using-answerai/sidekick-studio/sidekick-settings/general.md
Comment (maxtechera) [inline packages/docs/docs/using-answerai/sidekick-studio/sidekick-settings/general.md:1]: 🙌

File: packages/server/src/services/chatflows/index.ts [hunk 2]
```diff
@@ -349,18 +349,34 @@ const importChatflows = async (newChatflows: Partial<ChatFlow>[]): Promise<any>
...
+                existingChatbotConfig.displayMode = updatedConfig.displayMode
+            }
+            if (updatedConfig.embeddedUrl !== undefined) {
+                existingChatbotConfig.embeddedUrl = updatedConfig.embeddedUrl
+            }
+        }
+
+        // Update the chatbotConfig in the chatflow object
+        chatflow.chatbotConfig = JSON.stringify(existingChatbotConfig)
+
```
Comment (maxtechera) [inline packages/server/src/services/chatflows/index.ts:373]: why is this parse/unparse needed?

**Review by maxtechera — CHANGES_REQUESTED** [review]

File: packages/server/src/services/chatflows/index.ts (hunk 2 above)
Comment (bradtaylorsf) [reply packages/server/src/services/chatflows/index.ts:373]: The parse/unparse operation in the updateChatflow function is needed for a few reasons:

1. Type safety: The chatbotConfig is stored as a string in the database, but we need to work with it as an object in our code. Parsing it allows us to manipulate it as a JavaScript object.
2. Partial updates: The function allows for partial updates of the chatbotConfig. By parsing the existing config and the new config separately, we can selectively update only the fields that are provided in the new config.
3. Maintaining existing data: By parsing the existing chatbotConfig, we ensure that we don't lose any existing configuration data that isn't being updated in this operation.

**Pull request merged**